src/optlang/symbolics.py

```diff
@@ -120,21 +118,14 @@ def mul(*args):
     Mul = sympy.Mul
     Pow = sympy.Pow
 
-    class Symbol(sympy.Symbol):
+    class Symbol(sympy.core.Dummy):
         """A generic symbol used in expressions."""
 
-        def __new__(cls, name, **kwargs):
+        def __new__(cls, name, *args, **kwargs):
             if not isinstance(name, six.string_types):
                 raise TypeError("name should be a string, not %s" % repr(type(name)))
 
-            obj = sympy.Symbol.__new__(cls, str(uuid.uuid1()))
-
-            obj.name = name
-            obj._assumptions = FactKB(_assume_rules)
-            obj._assumptions._tell("commutative", True)
-            obj._assumptions._tell("uuid", uuid.uuid1())
```

Comment:
Are we not losing behavior by removing these?

Reply:
No. Commutative is set by default for sympy symbols. The uuid was the previous hack to make symbols unique. In sympy `x = Symbol(x); y = Symbol("x")` will have x and y point to the same object which would not work in optlang. In prior versions of sympy symbols would not be the same if they had different assumptions, so injecting a unique uuid into them made them unique, but that was exploiting some undefined behavior in sympy. 1.12 changed that behavior somewhat, so xreplace did not work anymore, which is the bug that we were hitting. The Dummy is a version of the Symbol class that is always unique (independent of assumptions).